Content.Server/MachineLinking/System/DeadMansSwitchSystem.cs

```diff
@@ -0,0 +1,127 @@
+        SubscribeLocalEvent<DeadMansSwitchComponent, DeadMansSwitchDisarmCompleteEvent>(OnSuccessfulDisarm);
+
+        SubscribeLocalEvent<DeadMansSwitchHolderComponent, MobStateChangedEvent>(OnMobStateChanged);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+        foreach (var holder in EntityQuery<DeadMansSwitchHolderComponent>())
+        {
+            if (holder.Switches.Count == 0)
+            {
+                RemComp<DeadMansSwitchHolderComponent>(holder.Owner);
+            }
+        }
+    }
+
+    private void OnInit(EntityUid uid, DeadMansSwitchComponent component, ComponentInit args)
+    {
+        _signalSystem.EnsureTransmitterPorts(uid, component.Port);
+    }
+
+    private void OnDropped(EntityUid uid, DeadMansSwitchComponent component, DroppedEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        if (!component.Armed)
+            return;
+
+        _signalSystem.InvokePort(uid, component.Port);
+        component.Armed = false;
+        if(EntityManager.TryGetComponent<DeadMansSwitchHolderComponent>(args.User, out var holderComp))
+            holderComp.Switches.Remove(component);
+        _popupSystem.PopupEntity(Loc.GetString("dead-mans-switch-let-go"), args.User, Filter.Entities(args.User));
+        args.Handled = true;
+    }
+
+    private void OnUsedInHand(EntityUid uid, DeadMansSwitchComponent component, UseInHandEvent args)
+    {
+        if (component.Armed == false)
+        {
+            var holder = EnsureComp<DeadMansSwitchHolderComponent>(args.User);
+            holder.Switches.Add(component);
+            component.Armed = true;
+            _popupSystem.PopupEntity(Loc.GetString("dead-mans-switch-armed"), args.User, Filter.Entities(args.User));
+        }
+        else
+        {
+            _signalSystem.InvokePort(uid, component.Port);
+            _popupSystem.PopupEntity(Loc.GetString("dead-mans-switch-let-go"), args.User, Filter.Entities(args.User));
+        }
+    }
+
+    private void OnGetVerbs(EntityUid uid, DeadMansSwitchComponent component, GetVerbsEvent<AlternativeVerb> args)
+    {
+        if (component.Armed)
+        {
+            args.Verbs.Add(new AlternativeVerb()
+            {
+                Text = Loc.GetString("verb-dead-mans-switch-disarm"),
+                Act = () =>
+                {
+                    _doAfterSystem.DoAfter(new DoAfterEventArgs(args.User, 15f, target: uid)
+                    {
+                        BreakOnStun = true,
+                        NeedHand = true,
+                        TargetFinishedEvent = new DeadMansSwitchDisarmCompleteEvent(args.User)
+                    });
+                }
+            });
+        }
+    }
+
+    private void OnMobStateChanged(EntityUid uid, DeadMansSwitchHolderComponent component, MobStateChangedEvent args)
+    {
+        if (args.CurrentMobState == DamageState.Critical || args.CurrentMobState == DamageState.Dead)
```

Comment:
Don't we still have incapacitated state that's for when you want to check for either of these?

Reply:
no idea